### PR TITLE
Add option to disable SSL

### PR DESCRIPTION
**Pull request by urso**

### CODE REVIEW

**Comment by ph** [comment]
Fixes https://github.com/logstash-plugins/logstash-input-lumberjack/issues/55

File: lib/logstash/inputs/lumberjack.rb [hunk 1]
```diff
@@ -48,7 +51,7 @@ def register
 
     @logger.info("Starting lumberjack input listener", :address => "#{@host}:#{@port}")
     @lumberjack = Lumberjack::Server.new(:address => @host, :port => @port,
```
Comment (jordansissel) [inline lib/logstash/inputs/lumberjack.rb]: We should do some extra validation -

1) If ssl is true, require both ssl_certificate and ssl_key
2) If ssl is false, reject the presence of ssl_certificate and ssl_key, or at least warn that they are not used in this configuration.

Comment (urso) [inline lib/logstash/inputs/lumberjack.rb]: this check is done by the ruby-lumberjack server implementation:
https://github.com/elastic/ruby-lumberjack/blob/master/lib/lumberjack/server.rb#L31

Comment (ph) [inline lib/logstash/inputs/lumberjack.rb]: @urso even if the test is done at the library level its better to add validations inside the plugin and raise a `LogStash::ConfigurationError`, its bit more user friendly. We hope to provide finer grained validation options for the attributes in the future.

Comment (urso) [inline lib/logstash/inputs/lumberjack.rb]: added check

File: lib/logstash/inputs/lumberjack.rb [hunk 2]
```diff
@@ -46,9 +49,15 @@ def register
     require "logstash/circuit_breaker"
     require "logstash/sized_queue_timeout"
 
+    ssl_configured = !(@ssl_certificate.nil? && @ssl_key.nil?)
```
Comment (ph) [inline lib/logstash/inputs/lumberjack.rb]: Can you move this logic into a predicate method and create a unit test for it?

```ruby
def ssl_configured?
  !(@ssl_certificate.nil? && @ssl_key.nil?)
end
``

File: lib/logstash/inputs/lumberjack.rb [hunk 3]
```diff
@@ -46,9 +49,15 @@ def register
...
+    @logger.warn("SSL Certificate will not be used") if ssl_configured && !@ssl
+    if @ssl && !ssl_configured
+      raise LogStash::ConfigurationError, "Certificate or Certificate Key not configured"
+    end
```
Comment (ph) [inline lib/logstash/inputs/lumberjack.rb]: Can you create a test to cover this case?

**Comment by ph** [comment]
@urso really minor issues, almost ready to merge!

**Comment by ph** [comment]
code LGTM, new tests passed.

This one fail but it related to our refactoring of the shutdown behavior and will resolve itself with the shutdown changes in #52 
````
 Failure/Error: lumberjack.run(queue)
     NoMethodError:
       undefined method `shutdown' for #<LogStash::Inputs::Lumberjack:0x32d418a9>
     # ./lib/logstash/inputs/lumberjack.rb:109:in `run'
     # ./spec/inputs/lumberjack_spec.rb:93:in `(root)'

**Comment by m4ce** [comment]
Is there any plan to merge this? I would also like to disable SSL.

**Comment by Tenischev** [comment]
Hello, could you please merge it into master?

**Comment by jordansissel** [comment]
The beats Input replaces this and has ssl disabled by default.

**Comment by Tenischev** [comment]
Hello @jordansissel. Yes, sure, beats has ssl disabled by default. But unfortunately, i could not migrate from logstash-forwarder to filebeats because of Go is not working in my environment. I see that "All checks have passed" and @ph said that pull request is "almost ready to merge!". So, i thought maybe you will be mercy and merge this pull request into master. Of course, i can take the master and these changes and build plugin locally (actually already did). But it will be nice if one merge could save time for many peoples (urso, m4ce, me and maybe someone as unlucky as i am in the future).

**Comment by jordansissel** [comment]
Logstash-forwarder has no option to disable TLS. Further,
logstash-forwarder can send data to the beats input, so you should be able
to point logstash-forwarder at the beats input.

**Comment by urso** [comment]
This is a 3 years old PR on a long deprecated input plugin adding a feature that is not even supported by logstash-forwarder. Let's close it :)

**Pull request closed without merging**